### PR TITLE
Feature/CCM-6245 TFSec scanning

### PR DESCRIPTION
**Pull request by jamesthompson26-nhs**
## Description

Adds a TFSec scan to the CI Commit workflow.

## Context

TFSec keeps our HCL sane and secure. Builds will fail where a commit introduces a new issue of medium or above severity.

I've also removed .github from the VSCode workspace as it's something we need to actively (and relatively frequently) work on.

## Type of changes

- [ ] Refactoring (non-breaking change)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would change existing functionality)
- [ ] Bug fix (non-breaking change which fixes an issue)

## Checklist

- [ ] I am familiar with the [contributing guidelines](../docs/CONTRIBUTING.md)
- [ ] I have followed the code style of the project
- [ ] I have added tests to cover my changes
- [ ] I have updated the documentation accordingly
- [ ] This PR is a result of pair or mob programming

---

## Sensitive Information Declaration

To ensure the utmost confidentiality and protect your and others privacy, we kindly ask you to NOT including [PII (Personal Identifiable Information) / PID (Personal Identifiable Data)](https://digital.nhs.uk/data-and-information/keeping-data-safe-and-benefitting-the-public) or any other sensitive data in this PR (Pull Request) and the codebase changes. We will remove any PR that do contain any sensitive information. We really appreciate your cooperation in this matter.

- [ ] I confirm that neither PII/PID nor sensitive data are included in this PR and the codebase changes.

### CODE REVIEW

File: scripts/terraform/terraform.mk [hunk 1]
```diff
@@ -69,7 +69,7 @@ terraform-install: # Install Terraform @Installation
 ${VERBOSE}.SILENT: \
 	_terraform \
 	clean \
-	terraform-apply \
+	terraform-apply \q
```
Comment (aidenvaines-bjss) [inline scripts/terraform/terraform.mk:72]: Is this intentional?

File: scripts/setup/setup.sh [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+function main() {
+
+  cd "$(git rev-parse --show-toplevel)"
+
+  run-setup
+}
+
+function run-setup() {
+
+  git clone https://github.com/asdf-vm/asdf.git $HOME/.asdf --branch v0.14.0 -c advice.detachedHead=false
```
Comment (aidenvaines-bjss) [inline scripts/setup/setup.sh:22]: This branch should be a global please

File: docs/Gemfile.lock [hunk 3]
```diff
@@ -29,6 +29,7 @@ GEM
     ffi (1.16.3)
     forwardable-extended (2.6.0)
     gemoji (4.1.0)
+    google-protobuf (3.25.3-arm64-darwin)
```
Comment (aidenvaines-bjss) [inline docs/Gemfile.lock:32]: you've locked all this to ARM, undo, if you want to build use an AMD64/x86 .devcontainer

**Review by aidenvaines-bjss — CHANGES_REQUESTED** [review]

File: docs/Gemfile.lock (hunk 3 above)
Comment (jamesthompson26-nhs) [reply docs/Gemfile.lock:32]: not intentional. Will remove.

**Review by sidnhs — APPROVED** [review]

**Review by sidnhs — APPROVED** [review]

File: scripts/setup/setup.sh [hunk 4]
```diff
@@ -0,0 +1,43 @@
+#!/usr/bin/env bash
+
+# WARNING: Please DO NOT edit this file! It is maintained in the Repository Template (https://github.com/nhs-england-tools/repository-template). Raise a PR instead.
```
Comment (aidenvaines-bjss) [inline scripts/setup/setup.sh:3]: this comment line is wrong
```suggestion
# WARNING: Please DO NOT edit this file! It is maintained in the Repository Template (https://github.com/NHSDigital/nhs-notify-repository-template). Raise a PR instead.
```

**Review by aidenvaines-bjss — CHANGES_REQUESTED** [review]

**Review by sidnhs — APPROVED** [review]

**Review by aidenvaines-bjss — APPROVED** [review]

**Pull request merged**